src/Zicht/Bundle/SolrBundle/Manager/SolrEntityManager.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * @copyright Zicht Online <https://zicht.nl>
+ */
+
+namespace Zicht\Bundle\SolrBundle\Manager;
+
+use Zicht\Bundle\SolrBundle\Manager\Doctrine\SearchDocumentRepository;
+use Zicht\Bundle\SolrBundle\Solr\Client;
+use Zicht\Bundle\SolrBundle\Solr\QueryBuilder;
+
+/**
+ * As opposed to the regular SolrManager this SolrEntityManager keeps track of the handled objects to not handle the same
+ * object more than once per request. Also the SolrEntityManager checks if an entity has indexable related entities and
+ * runs an update on those too.
+ *
+ * @example To use the SolrEntityManager instead of the default SolrManager, add to your app/config/bundles/zicht_solr.yml:
+ * zicht_solr:
+ *     manager:
+ *         type: 'entity'
+ */
+class SolrEntityManager extends SolrManager
+{
+    /** @var string[] */
+    private $updatedEntityHashes = [];
+
+    /** @var string[] */
+    private $deletedEntityHashes = [];
+
+    /**
+     * @param mixed $entity
+     * @return bool
+     */
+    protected static function validateEntity($entity)
+    {
+        if (!is_object($entity)) {
+            $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2);
```

Comment:
`debug_backtrace` in productie code? Is vast heel handig voor leesbare foutmelding, maar weten we zeker dat dit op productie omgeving ook goed blijft werken?

Reply:
Zeker. Dit is een vanilla PHP functie en die maakt geen onderscheid tussen de Symfony environment. Weet ook niet waarom ze die `debug_` hebben geprefixt, want hij geeft gewoon een trace stack terug. Juist heel waardevol in je logs op productie, want op andere omgevingen kun je het zelf wel debuggen...